src/test/java/software/amazon/encryption/s3/S3EncryptionClientMatDescTest.java

```diff
@@ -0,0 +1,338 @@
+        .put("version", "1.0")
+        .put("admin", "yes")
+        .build())
+      .build();
+
+    S3Client wrappedClient= S3Client.create();
+    S3EncryptionClient client = S3EncryptionClient.builder()
+      .keyring(rsaKeyring)
+      .instructionFileConfig(InstructionFileConfig.builder()
+        .enableInstructionFilePutObject(true)
+        .instructionFileClient(wrappedClient)
+        .build())
+      .build();
+
+    final String input = "Testing Materials Description in Instruction File!";
+    final String objectKey = appendTestSuffix("test-rsa-materials-description-in-instruction-file");
+
+    client.putObject(builder -> builder
+      .bucket(BUCKET)
+      .key(objectKey)
+      .build(), RequestBody.fromString(input)
+    );
+    ResponseBytes<GetObjectResponse> responseBytes = client.getObjectAsBytes(builder -> builder
+      .bucket(BUCKET)
+      .key(objectKey)
+      .build());
+    assertEquals(input, responseBytes.asUtf8String());
+
+    S3Client defaultClient= S3Client.create();
+
+    ResponseBytes<GetObjectResponse> directInstGetResponse = defaultClient.getObjectAsBytes(builder -> builder
+      .bucket(BUCKET)
+      .key(objectKey + ".instruction")
+      .build());
+
+    String instructionFileContent = directInstGetResponse.asUtf8String();
+    JsonNodeParser parser = JsonNodeParser.create();
+    JsonNode instructionFileNode = parser.parse(instructionFileContent);
+
+    JsonNode matDescNode = parser.parse(instructionFileNode.asObject().get("x-amz-matdesc").asString());
+    assertEquals("1.0", matDescNode.asObject().get("version").asString());
+    assertEquals("yes", matDescNode.asObject().get("admin").asString());
+
+    deleteObject(BUCKET, objectKey, client);
+  }
+
+  @Test
+  public void testAesKeyringMatDescOverridesPutObjectEncryptionContext() {
+    AesKeyring aesKeyring = AesKeyring.builder()
+      .wrappingKey(AES_KEY)
+      .secureRandom(new SecureRandom())
+      .materialsDescription(MaterialsDescription.builder()
+        .put("version", "1.0")
+        .build())
+      .build();
+
+    S3Client wrappedClient= S3Client.create();
+    S3EncryptionClient client = S3EncryptionClient.builder()
+      .keyring(aesKeyring)
+      .instructionFileConfig(InstructionFileConfig.builder()
+        .enableInstructionFilePutObject(true)
+        .instructionFileClient(wrappedClient)
+        .build())
+      .build();
+
+    final String input = "Testing Materials Description in Instruction File and not Encryption Context!";
+    final String objectKey = appendTestSuffix("test-aes-materials-description-in-instruction-file-and-not-encryption-context");
+    final String encryptionContext = "{\"admin\":\"yes\"}";
+
+    client.putObject(builder -> builder
+      .bucket(BUCKET)
+      .key(objectKey)
+      .serverSideEncryption(ServerSideEncryption.AWS_KMS)
+      .ssekmsEncryptionContext(Base64.getEncoder().encodeToString(encryptionContext.getBytes(StandardCharsets.UTF_8)))
```

Comment:
This is not how EncryptionContext is passed to S3EC. 

Refer to e.g. `KmsContextV2toV3` in the compatibility tests for an example of S3EC EncryptionContext.